Makefile

```diff
@@ -0,0 +1,96 @@
+GO_TEST_CODECOV ?=
+
+TOP_DIR ?= $(shell git rev-parse --show-toplevel)
+VERSION ?= $(shell git describe --abbrev=0)
+BUILD ?= $(shell date +%FT%T%z)
+GOVERSION ?= $(shell go version | cut -d " " -f3)
+COMMIT ?= $(shell git rev-parse HEAD)
+BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+GO_BUILD_LDFLAGS = -X main.Version=${VERSION} -X main.Build=${BUILD} -X main.Commit=${COMMIT} -X main.Branch=${BRANCH} -X main.GoVersion=${GOVERSION} -s -w
+NAME ?= mnogo_exporter
+REPO ?= percona/$(NAME)
+GORELEASER_FLAGS ?=
+UID ?= $(shell id -u)
+
+export TEST_PSMDB_VERSION?=3.6
+export TEST_MONGODB_FLAVOR?=percona/percona-server-mongodb
+export TEST_MONGODB_ADMIN_USERNAME?=
+export TEST_MONGODB_ADMIN_PASSWORD?=
+export TEST_MONGODB_USERNAME?=
+export TEST_MONGODB_PASSWORD?=
+export TEST_MONGODB_S1_RS?=rs1
+export TEST_MONGODB_STANDALONE_PORT?=27017
+export TEST_MONGODB_S1_PRIMARY_PORT?=17001
+export TEST_MONGODB_S1_SECONDARY1_PORT?=17002
+export TEST_MONGODB_S1_SECONDARY2_PORT?=17003
+export TEST_MONGODB_S2_RS?=rs2
+export TEST_MONGODB_S2_PRIMARY_PORT?=17004
+export TEST_MONGODB_S2_SECONDARY1_PORT?=17005
+export TEST_MONGODB_S2_SECONDARY2_PORT?=17006
+export TEST_MONGODB_CONFIGSVR_RS?=csReplSet
+export TEST_MONGODB_CONFIGSVR1_PORT?=17007
+export TEST_MONGODB_CONFIGSVR2_PORT?=17008
+export TEST_MONGODB_CONFIGSVR3_PORT?=17009
+export TEST_MONGODB_MONGOS_PORT?=17000
+
+define TEST_ENV
+	TEST_MONGODB_ADMIN_USERNAME=$(TEST_MONGODB_ADMIN_USERNAME) \
+	TEST_MONGODB_ADMIN_PASSWORD=$(TEST_MONGODB_ADMIN_PASSWORD) \
+	TEST_MONGODB_USERNAME=$(TEST_MONGODB_USERNAME) \
+	TEST_MONGODB_PASSWORD=$(TEST_MONGODB_PASSWORD) \
+	TEST_MONGODB_S1_RS=$(TEST_MONGODB_S1_RS) \
+	TEST_MONGODB_STANDALONE_PORT=$(TEST_MONGODB_STANDALONE_PORT) \
+	TEST_MONGODB_S1_PRIMARY_PORT=$(TEST_MONGODB_S1_PRIMARY_PORT) \
+	TEST_MONGODB_S1_SECONDARY1_PORT=$(TEST_MONGODB_S1_SECONDARY1_PORT) \
+	TEST_MONGODB_S1_SECONDARY2_PORT=$(TEST_MONGODB_S1_SECONDARY2_PORT) \
+	TEST_MONGODB_S2_RS=$(TEST_MONGODB_S2_RS) \
+	TEST_MONGODB_S2_PRIMARY_PORT=$(TEST_MONGODB_S2_PRIMARY_PORT) \
+	TEST_MONGODB_S2_SECONDARY1_PORT=$(TEST_MONGODB_S2_SECONDARY1_PORT) \
+	TEST_MONGODB_S2_SECONDARY2_PORT=$(TEST_MONGODB_S2_SECONDARY2_PORT) \
+	TEST_MONGODB_CONFIGSVR_RS=$(TEST_MONGODB_CONFIGSVR_RS) \
+	TEST_MONGODB_CONFIGSVR1_PORT=$(TEST_MONGODB_CONFIGSVR1_PORT) \
+	TEST_MONGODB_CONFIGSVR2_PORT=$(TEST_MONGODB_CONFIGSVR2_PORT) \
+	TEST_MONGODB_CONFIGSVR3_PORT=$(TEST_MONGODB_CONFIGSVR3_PORT) \
+	TEST_MONGODB_MONGOS_PORT=$(TEST_MONGODB_MONGOS_PORT) \
+	TEST_PSMDB_VERSION=$(TEST_PSMDB_VERSION) \
+	TEST_MONGODB_FLAVOR=$(TEST_MONGODB_FLAVOR)
+endef
+
+env:
+	@echo $(TEST_ENV) | tr ' ' '\n' >.env
+
+init:                       ## Install linters.
+	curl https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s
+	curl https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s latest
+	go get golang.org/x/tools/cmd/goimports
+	# To prevent goimports being marked as dependency
+	go mod tidy
```

Comment:
On SaaS we download all such tools as sources and use `tools.go` to pin versions
```
init:                   ## Install development tools
	go build -modfile=tools/go.mod -o bin/goimports golang.org/x/tools/cmd/goimports
	go build -modfile=tools/go.mod -o bin/golangci-lint github.com/golangci/golangci-lint/cmd/golangci-lint
	go build -modfile=tools/go.mod -o bin/reviewdog github.com/reviewdog/reviewdog/cmd/reviewdog
```

It guarantees that each developer has same tools versions. If you want I can make a PR to show how we use that approach.

Reply:
Yes please.

Reply:
Here it is: https://github.com/Percona-Lab/mnogo_exporter/pull/5